compiler/ballerina-lang/src/main/java/io/ballerina/projects/plugins/completion/AbstractCompletionProvider.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (http://wso2.com) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.ballerina.projects.plugins.completion;
+
+import io.ballerina.compiler.syntax.tree.Node;
+
+import java.util.List;
+
+/**
+ * Interface for completion item providers.
+ *
+ * @param <T> Provider's node type
+ * @since 2201.7.0
+ */
+public abstract class AbstractCompletionProvider<T extends Node> implements CompletionProvider<T> {
+
+    private final List<Class<T>> attachmentPoints;
+
+    public AbstractCompletionProvider(List<Class<T>> attachmentPoints) {
```

Comment:
Should we take the attachment points (or supported nodes) as a constructor parameter? Can't the class names be hardcoded in the `getAttachmentPoints()` method itself? Because the completion provider class knows exactly what it supports completions for.

Reply:
Removed the AbstractCompletionProvider class and hardcoded the supported nodes in the getAttachmentPoints method